ceph/rados_utils.py

```diff
@@ -219,10 +219,27 @@ def check_osd_up(self, osd_id):
         """
         end_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
         while end_time > datetime.datetime.now():
-            status = self.is_up(osd_id)
-            if status:
-                return 1
             time.sleep(5)
+            if not self.is_up(osd_id):
+                print("OSD is down")
+                return 0
+        return 1
+
+    def wait_osd_up(self, osd_id):
+        """
+        Checks if the given osd is up and if the osd is down, waits for 120 seconds for the same to come up
+        Args:
+            osd_id: ID of the osd to be checked
+
+        Returns: 1 if up, 0 if down
+
+        """
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
+        while end_time > datetime.datetime.now():
+            time.sleep(5)
+            if self.is_up(osd_id):
+                print("OSD is up")
+                return 1
```

Comment:
IMHO, we can achieve this with a single method `def wait_until(self, osd, up=True) -> bool:` as the difference is the state of OSD daemon.  For example
```
def wait_until_osd_state(self, osd, down=True) -> bool:
  end_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
  while end_time > datetime.datetime.now():
    time.sleep(5)
     osd_is_up = self.is_up(osd_id):
     if osd_is_up and not down: return True
     if down and not osd_is_up: return True

  return False
```

Reply:
Added a single method.